sdk/iot/provisioning/src/az_iot_provisioning_client_sas.c

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_iot_provisioning_client.h>
+#include <az_precondition.h>
+#include <az_precondition_internal.h>
+#include <az_span.h>
+#include <az_span_internal.h>
+
+#include <stdint.h>
+
+#include <_az_cfg.h>
+
+#define LF '\n'
+#define AMPERSAND '&'
+#define EQUAL_SIGN '='
+#define SCOPE_REGISTRATIONS_STRING "%2fregistrations%2f"
+#define SAS_TOKEN_SR "SharedAccessSignature sr"
+#define SAS_TOKEN_SE "se"
+#define SAS_TOKEN_SIG "sig"
+#define SAS_TOKEN_SKN "skn"
+
+static const az_span resources_string = AZ_SPAN_LITERAL_FROM_STR(SCOPE_REGISTRATIONS_STRING);
+static const az_span sr_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SR);
+static const az_span sig_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SIG);
+static const az_span skn_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SKN);
+static const az_span se_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SE);
+
+AZ_NODISCARD az_result az_iot_provisioning_client_sas_get_signature(
+    az_iot_provisioning_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
+    az_span* out_signature)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION(token_expiration_epoch_time > 0);
+  AZ_PRECONDITION_VALID_SPAN(signature, 0, false);
+  AZ_PRECONDITION_NOT_NULL(out_signature);
+
+  // Produces the following signature:
+  // url-encoded(<resource-string>)\n<expiration-time>
+  // Where
+  // resource-string: <scope-id>/registrations/<registration-id>
+
+  int32_t required_size = az_span_size(client->_internal.id_scope)
+      + az_span_size(client->_internal.registration_id) + az_span_size(resources_string)
+      + u32toa_size(token_expiration_epoch_time)
+      + 1; // 1 is for one line feed character.
```

Comment:
>1 [](start = 8, length = 1)

This is little different from what you did is iothub_client.
Lets make it similar?

+ (uint32_t) sizeof(LF):

Reply:
I changed everything to numbers and added comments on the sides. On both dps and hub clients.

Reply:
Done. All changed to literals.